azurerm/resource_arm_cosmosdb_sql_container_test.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMCosmosDbSqlContainer_basic(t *testing.T) {
+	ri := tf.AccRandTimeInt()
+	resourceName := "azurerm_cosmosdb_sql_container.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMCosmosDbSqlContainerDestroy,
+		Steps: []resource.TestStep{
+			{
+
+				Config: testAccAzureRMCosmosDbSqlContainer_basic(ri, testLocation()),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testCheckAzureRMCosmosDbSqlContainerExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMCosmosDbSqlContainerDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).cosmosAccountsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_cosmosdb_sql_container" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		account := rs.Primary.Attributes["account_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		database := rs.Primary.Attributes["database"]
+
+		resp, err := client.GetSQLContainer(ctx, resourceGroup, account, database, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Error checking destroy for Cosmos SQL Container %s (account %s) still exists:\n%v", name, account, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Cosmos SQL Container %s (account %s) still exists:\n%#v", name, account, resp)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMCosmosDbSqlContainerExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*ArmClient).cosmosAccountsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		account := rs.Primary.Attributes["account_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		database := rs.Primary.Attributes["database"]
+
+		resp, err := client.GetSQLContainer(ctx, resourceGroup, database, account, name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on cosmosAccountsClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Cosmos Container '%s' (account: '%s') does not exist", name, account)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMCosmosDbSqlContainer_basic(rInt int, location string) string {
+	res := fmt.Sprintf(`
```

Comment:
Could we change this to
```suggestion
	return fmt.Sprintf(`
```